jpa/eclipselink.jpa.test.jse/src/org/eclipse/persistence/jpa/test/query/TestQueryCase.java

```diff
@@ -16,236 +17,290 @@
+            if (!POPULATED)
+                populate();
 
-        EntityManager em = emf.createEntityManager();
+            em = emf.createEntityManager();
 
-        em.getTransaction().begin();
+            em.getTransaction().begin();
 
-        TypedQuery<EntityTbl01> query = em.createQuery(""
-                + "SELECT t FROM EntityTbl01 t "
+            TypedQuery<EntityTbl01> query = em.createQuery(""
+                    + "SELECT t FROM EntityTbl01 t "
                     + "WHERE t.itemString1 = ( "
-                        + "CASE t.itemInteger1 "
-                            + "WHEN 1000 THEN '047010' "
-                            + "WHEN 100 THEN '023010' "
-                            + "ELSE '033020' "
-                        + "END )", EntityTbl01.class);
-
-        List<EntityTbl01> dto01 = query.getResultList();
-        assertNotNull(dto01);
-        assertEquals(0, dto01.size());
-
-        query = em.createQuery(""
-                + "SELECT t FROM EntityTbl01 t "
+                    + "CASE t.itemInteger1 "
+                    + "WHEN 1000 THEN '047010' "
+                    + "WHEN 100 THEN '023010' "
+                    + "ELSE '033020' "
+                    + "END )", EntityTbl01.class);
+
+            List<EntityTbl01> dto01 = query.getResultList();
+            assertNotNull(dto01);
+            assertEquals(0, dto01.size());
+
+            query = em.createQuery(""
+                    + "SELECT t FROM EntityTbl01 t "
                     + "WHERE t.itemString1 = ( "
-                        + "CASE t.itemInteger1 "
-                            + "WHEN 1 THEN 'A' "
-                            + "WHEN 100 THEN 'B' "
-                            + "ELSE 'C' "
-                        + "END )", EntityTbl01.class);
-        dto01 = query.getResultList();
-        assertNotNull(dto01);
-        assertEquals(1, dto01.size());
-
-        assertEquals("A", dto01.get(0).getItemString1());
-        assertEquals("B", dto01.get(0).getItemString2());
-        assertEquals("C", dto01.get(0).getItemString3());
-        assertEquals("D", dto01.get(0).getItemString4());
-        assertEquals(new Integer(1), dto01.get(0).getItemInteger1());
-
-        em.getTransaction().rollback();
+                    + "CASE t.itemInteger1 "
+                    + "WHEN 1 THEN 'A' "
+                    + "WHEN 100 THEN 'B' "
+                    + "ELSE 'C' "
+                    + "END )", EntityTbl01.class);
+            dto01 = query.getResultList();
+            assertNotNull(dto01);
+            assertEquals(1, dto01.size());
+
+            assertEquals("A", dto01.get(0).getItemString1());
+            assertEquals("B", dto01.get(0).getItemString2());
+            assertEquals("C", dto01.get(0).getItemString3());
+            assertEquals("D", dto01.get(0).getItemString4());
+            assertEquals(new Integer(1), dto01.get(0).getItemInteger1());
+        } catch (Exception e) {
+            fail(e.getLocalizedMessage());
```

Comment:
I would probably avoid adding this explicit catch and just add finally - in case of an exception, it will bubble up. With  proposed change, full stacktrace is lost and one has to re-run updated the test to get it

Reply:
I agree. I've been trying to phase out the `catch` blocks from the JSE bucket since @jgrassel brought up the same issue. Also, I'm not sure I understand why an `em != null` check is needed at the bottom. If you use the structure I referenced in my other review comment, `em` cannot be null within the finally block.

Reply:
@dazey3 we've been discussing this with @rfelcman today and issues he is seeing happen if and only if TestQueryCase runs before others from the same package. When it runs last, no failure/issue is being observed - so the problem is related to the test execution order.

btw: when going through the test class itself, I also suggested to simplify it - it's JUnit 4 after all, so one can use Before/After to avoid copy/pasting stuff in tests (ie calls to populate, em creation etc)